server/src/main/java/io/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -94,6 +96,8 @@ public CoordinatorDynamicConfig(
                                    ? current.getKillDataSourceWhitelist()
                                    : parseKillDataSourceWhitelist(killDataSourceWhitelist);
 
+    this.maxSegmentsInQueue = maxSegmentsInQueue == null ? current.getMaxSegmentsInQueue() : maxSegmentsInQueue;
```

Comment:
@leiwangx I think widely used convention is just 0=unlimited. Negative values could be result of typo and should be rejected IMO.